src/main/java/com/example/demo/DemoApplication.java

```diff
@@ -1,26 +1,31 @@
 package com.example.demo;
 
+import com.example.demo.matcher.InitialiseFakeMarket;
+import com.example.demo.matcher.Matcher;
 import com.example.demo.security.service.UserService;
 import com.example.demo.security.userInfo.AppUser;
 import com.example.demo.security.userInfo.Role;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.annotation.Bean;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
-import org.springframework.security.crypto.password.PasswordEncoder;
 
+import javax.persistence.Tuple;
 import java.util.ArrayList;
 
-@SpringBootApplication
+@SpringBootApplication @AllArgsConstructor
 public class DemoApplication {
 
+	@Autowired
+	private final Matcher matcher;
+
 	public static void main(String[] args) {
 		SpringApplication.run(DemoApplication.class, args);
 	}
 
-	@Bean
-		// this interface is used to run a code block exactly once upon initialisation of the program
+	@Bean// this bean runs a code block exactly once upon initialisation of the program
 	CommandLineRunner run (UserService userService) {
```

Comment:
To avoid adding a constructor to this class you could pass the matcher as an argument just like you did for the user service. I believe Springboot should do the work for you and you will no longer need the autowired matcher either.

Reply:
good spot